### PR TITLE
HKDs

### PR DESCRIPTION
**Pull request by AriFordsham**

### CODE REVIEW

File: src/Hedgehog/Plutus/TestData.hs [hunk 1]
```diff
@@ -138,26 +140,63 @@ instance (Integral i) => TestData (ShouldBeNatural i) where
 
   generalise = MightBeNegative . fromIntegral
 
-newtype Good' a = Good {unGood' :: Good a}
+newtype Good' a = G {unGood' :: Good a}
 
 deriving stock instance (Eq (Good a)) => Eq (Good' a)
 deriving stock instance (Show (Good a)) => Show (Good' a)
 
+newtype HKD d = HKD (d I)
```
Comment (Geometer1729) [inline src/Hedgehog/Plutus/TestData.hs:148]: Afaict there's no reason to have this newtype. I've removed it on `brian/ditch-newtype` and it seems to be fine.

**Review by Geometer1729 — CHANGES_REQUESTED** [review]
I think the HKD newtype should be removed, but otherwise LGTM

File: src/Hedgehog/Plutus/TestData.hs (hunk 1 above)
Comment (AriFordsham) [reply src/Hedgehog/Plutus/TestData.hs:148]: It's for deriving via.

**Review by Geometer1729 — APPROVED** [review]

File: src/Hedgehog/Plutus/TestData.hs (hunk 1 above)
Comment (Geometer1729) [reply src/Hedgehog/Plutus/TestData.hs:148]: Oh that makes sense, sorry about that.

Comment (AriFordsham) [reply src/Hedgehog/Plutus/TestData.hs:148]: It's not considered good practice to declare instances for top-level type variables.

**Pull request merged**